src/GitHubTools/Functions/Public/Update-AzurePipelineDependencies.ps1

```diff
@@ -0,0 +1,118 @@
+            if ($_.ref -like "refs/tags/*") {
+                $TagName = $_.ref.replace("refs/tags/", "")
+                $DependencyOrg = $_.name.Split("/")[0]
+                $DependencyRepo = $_.name.Split("/")[1]
+                $Releases = Get-GitHubRepoRelease -GitHubOrganisation $DependencyOrg -RepositoryName $DependencyRepo
+                $OutOfDate = $Releases[0].tag_name -ne $TagName
+
+                if ($OutOfDate) {
+                    Write-Warning "$($_.name) dependency: $TagName should be $($Releases[0].tag_name): updating"
+                    $PipelineYml.Content = $PipelineYml.Content.Replace($TagName, $Releases[0].tag_name)
+                    $UpdatePipelineYml = $true
+                }
+                else {
+                    Write-Host "$($_.name) dependency up to date" -ForegroundColor Green
+                }
+            }
+            else {
+                Write-Warning "No tag reference for: $($_.name) - $($_.ref)"
+                return
+            }
+        }
+
+        $PipelineYml.Content = $PipelineYml.Content.Replace("      - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks", "")
+
+        $CodeBuildYml = Get-GitHubRepoFileContent -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -FilePath "pipeline-templates/job/code-build.yml"
+
+        $UpdateCodeBuildYml = $false
+        if ($CodeBuildYml) {
+            if ($CodeBuildYml.Content.Contains("  - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks")) {
+                $CodeBuildYml.Content = $CodeBuildYml.Content.Replace("  - template: azure-pipelines-templates/build/step/dependency-check.yml@das-platform-building-blocks", "")
+                $UpdateCodeBuildYml = $true
+            }
+        }
+        if ($UpdatePipelineYml || $UpdateCodeBuildYml) {
+            $NewBranchName = "pipeline-dependency-updates-" + (Get-Date -Format "FileDate")
+
+            $DefaultBranchRef = Get-GithubRepoBranchRef -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name -BranchName $_.defaultBranchRef.name
+
+            New-GitHubRepoBranch -GitHubOrganisation $GitHubOrganisation -RepositoryName $Repo.name `
+                -BaseRefSha $DefaultBranchRef.object.sha -NewBranchName $NewBranchName
+
+            $FileContentParams = @{
+                GitHubOrganisation = $GitHubOrganisation
+                RepositoryName     = $Repo.Name
+                BranchName         = $NewBranchName
+                FilePath           = "azure-pipelines.yml"
+                FileContent        = $PipelineYml.Content
+                CommitMessage      = "Updating pipeline dependencies"
+                BaseRefSha         = $PipelineYml.Sha
+            }
+            Set-GitHubRepoFileContent @FileContentParams
+
+            if ($UpdateCodeBuildYml) {
+                $CodeBuildYml.Content
```

Comment:
Is this for logging? Should it be Write-Host/Verbose?

Reply:
Good spot, removed